kent_a07b.js

```diff
@@ -0,0 +1,101 @@
+        console.log(getPrompt(groceryItems)); // logging prompt to terminal -- receiving prompt from invoking function getPrompt
+    }
+} 
+    //Step 4 End
+});
+
+/* DEFINE YOUR FUNCTIONS BETWEEN THIS LINE */
+
+//Step 2
+function buildFoodsList (badFoods, ...moreBadFoods){ //declaring function with array parameter and rest operator parameter to accept any number of arguments
+    for(const food of moreBadFoods){ //for each loop to iterate each input argument because of the rest operator
+        badFoods.push(food); //using push method to add each argument to badFoods array
+    }
+}
+
+function getPrompt(groceryItems){ //declaring function with single parameter
+    let message = `Your grocery list contains ${groceryItems.length} ${groceryItems.length === 1 ? "item" : "items" } 
+Please enter a grocery item.
+Enter Q to quit.
+Enter V to view list.
+========================`; //declared variable message with template literal to accept array length -- also used ternary operator to decide between singular and plural depending on number of items in array
+    return message; //returning string value 
+} //is there a specific reason that the getPrompt and displayList functions should return strings rather than logging through the function?
+
+function displayList(groceryItems){ //declaring function to display groceryItems array - single parameter needed
+    let eqSign = "="; //declaring and intializing to string value to make the message aesthetics more simple
+    let listMessage = ""; //declaring and initializing empty string variable to accept list items upon loop iteration
+    for (let i = 0; i < groceryItems.length; i++){ //for loop to add each value of array to listMessage
+        listMessage += (i+1) + "). " + groceryItems[i] + "\n"; //formatting and adding each list entry to listMessage 
+    }
+    let message = `Grocery List
+(${groceryItems.length} ${groceryItems.length === 1 ? "item" : "items"})
+${eqSign.repeat(25)}
+${listMessage}`; //declaring and initializing message variable as template literal 
+return message; //returning string value of variable message
+}
+//Step 3
+function checkItem(groceryItem, badFoods){ //declaring function to check if item entered is on the badFoods array
+    let boolFlag = false; //declaring and initializing variable to store false value from array iteration
+    for(const food of badFoods){ //foreach loop to iterate each index of badFoods
+        if (food.toLowerCase() === groceryItem.toLowerCase()){ //checking the entered item against this iteration of array
+            return true; //in the case that the item is in array -- returning true to end function
+        }else { //if the check doesn't return true the else statement sets the flag variable to false again
+            boolFlag = false; //resetting value to false -- is it bad practice to have redundant code in here, or and empty else statement?
+        }
+    }
+    return boolFlag; //returning boolean value
```

Comment:
you've got the return here - but you also return in the if. I'd say just either pick one and be consistent. Return in the if/else - or set the flag and return after.